### DISCUSSION TITLE
Filtering Courts by State and level

### DISCUSSION THREAD

**Discussion by krward2** [freelawproject/courtlistener#3303] [Q&A] [answered]
I would like to get a list of Justice of the Peace and Municipal courts in the state of Texas. Is there a way to filter by state, and then by a type of some sort resembling county and municipal courts?

**Comment by flooie** [comment, marked as answer]
@krward2 Hi Kenneth  

Courtlistener does not have municipal Texas court data in it.  If it did and its as organized to link all of the Justice of the Peace courts and municipal courts to a centralized Parent Court - see model below you could make this query

Court.objects.get(id="txjustpeacect").child_courts.all()
Court.objects.get(id="texmunict").child_courts.all()

Assuming those were the IDs for those courts.  But since the data isnt in our system you cant do it.  

There would be a similar filtering mechanism in the API, but we have not yet introduced it into the court filter because we only recently added model changes to the db.

```
    parent_court = models.ForeignKey(
        "self",
        help_text="Parent court for subdivisions",
        on_delete=models.SET_NULL,
        blank=True,
        null=True,
        related_name="child_courts",
    )
```

I hope that helps.
